### ISSUE TITLE
Read the path files correctly when setting new default workspace and VALIDATE_ALL_CODEBASE: false 

### ISSUE THREAD

**Issue by tomkortjak** [oxsecurity/megalinter#2912]
We want to make it possible to configure megalinter to lint only certain projects in a monorepo, without scanning the entire codebase. The problems as described below is now causing  line 761 in MegaLinter.py to not read out the files correctly causing it to filter out modified files that should have been linted.

### Discussed in https://github.com/oxsecurity/megalinter/discussions/2906

How to configure mega-linter to only lint a certain folder in monorepo?

<div type='discussions-op-text'>

<sup>Originally posted by **tomkortjak** August 22, 2023</sup>
my configuration is as follows:
```
FAIL_IF_MISSING_LINTER_IN_FLAVOR: true
APPLY_FIXES: all
ENABLE_LINTERS:
  - JAVASCRIPT_ES
  - TYPESCRIPT_ES
  - TSX_ESLINT
  - JSX_ESLINT
#DISABLE:
#  - COPYPASTE
#  - SPELL 
SHOW_ELAPSED_TIME: true
PRINT_ALPACA: false
FILEIO_REPORTER: false
# DISABLE_ERRORS: true
FLAVOR_SUGGESTIONS: true 
VALIDATE_ALL_CODEBASE: false 
LOG_LEVEL: DEBUG
LINTER_RULES_PATH: .
```

workflow job: 
```
- name: MegaLinter
        id: ml
        uses: oxsecurity/megalinter/flavors/javascript@v7
        env: 
          DEFAULT_WORKSPACE: app/react
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          MEGALINTER_CONFIG: .github/.mega-linter.yml
```

The project contains a react project in app/react. I basically want to linter to only check for modified files in this directory. When scanning for modified files a get a correct list from the git diff, but the filtered list only includes the .gitignore:
```
Listing updated files in [/github/workspace] using git diff.
  Failed checking if running in CYGWIN due to: FileNotFoundError(2, 'No such file or directory')
  Popen(['git', 'fetch', 'origin', 'HEAD:refs/remotes/origin/HEAD'], cwd=/github/workspace, universal_newlines=False, shell=None, istream=None)
  Popen(['git', 'diff', '--name-only', 'origin/HEAD'], cwd=/github/workspace, universal_newlines=False, shell=None, istream=None)
  Modified files:
  .github/.mega-linter.yml
  .github/workflows/react_tests.yml
  .gitignore
  app/react/.cspell.json
  app/react/.eslintrc.js
  app/react/.jscpd.json
  app/react/npm-shrinkwrap.json
  app/react/src/pages/administration/inbox/InboxEditForm.tsx
  All found files before filtering:
  - .gitignore
  - File extensions: .js, .jsx, .ts, .tsx, .vue
```

I suspect it's because of line 761 in MegaLinter.py
`  if os.path.isfile(self.workspace + os.path.sep + diff_line): `

How should I configure the workflow so that it recognises files like "app/react/src/pages/administration/inbox/InboxEditForm.tsx" ?

Additional question, currently we are setting DEFAULT_WORKSPACE: app/react. Otherwise it can't find the .eslintrc.json file. Is there also another to configure the path to this lint file? I wasn't able to find any in the docs. </div>

**Comment by aiyajkwebelight** [comment]
+1, facing same.

setting default workplace explicitly with VALIDATE_ALL_CODEBASE: false not working, however VALIDATE_ALL_CODEBASE: true working fine.

**Comment by Kurt-von-Laven** [comment]
Have you already tried setting [`FILTER_REGEX_INCLUDE`](https://megalinter.io/latest/config-filtering/) and/or [`LINTER_RULES_PATH`](https://megalinter.io/latest/config-variables/)?

**Comment by tomkortjak** [comment]
> Have you already tried setting [`FILTER_REGEX_INCLUDE`](https://megalinter.io/latest/config-filtering/) and/or [`LINTER_RULES_PATH`](https://megalinter.io/latest/config-variables/)?

Yes both I have tried

**Comment by Kurt-von-Laven** [comment]
This is probably wishful thinking, but two changes that may be related to this issue were released in MegaLinter v7.4.0. #2964 modified the line you mentioned, and #2917 modified a similar line.

**Comment by tomkortjak** [comment]
Just tried 7.4, sadly hasn't fixed it yet.
